src/Adapter/AdapterAbstractServiceFactory.php

```diff
@@ -58,19 +46,14 @@ public function canCreateServiceWithName(ServiceLocatorInterface $serviceLocator
     public function __invoke(ContainerInterface $container, $requestedName, ?array $options = null)
     {
         $config = $this->getConfig($container);
-        return new Adapter($config[$requestedName]);
-    }
 
-    /**
-     * Create service with name
-     *
-     * @param string $name
-     * @param string $requestedName
-     * @return Adapter
-     */
-    public function createServiceWithName(ServiceLocatorInterface $serviceLocator, $name, $requestedName)
-    {
-        return $this($serviceLocator, $requestedName);
+        return new Adapter(
+            $config[$requestedName],
+            $container->get(Driver\DriverInterface::class),
+            $container->get(Platform\PlatformInterface::class),
```

Comment:
It might be worth refactoring these two to be similar to the following to, in that they check if a service is available first, doing that before instantiating and returning the Adapter object.